### ISSUE TITLE
Disable or remove specific code action/fixes/refactoring

### ISSUE THREAD

**Issue by heroboy** [microsoft/TypeScript#37332]
## Suggestion
https://github.com/microsoft/vscode/issues/92305

Because when some code action is available, there will be a icon shows on the code editing area.
If some fixes I always not want, it's annoying that the clickable icon will show every time my cursor move on the code or select some code or some other actions. Especially when the current line left is not white spaces, the icon will show at previous line and cover the privious line code.

For example, I always want use require in js file not import.

And maybe some visual effect like the dash underline, the show fixes button in the popup tip, the dimmed color text should be removed, when the relate code action is disabled. Because these hint there are problems in my code but actually not.

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by RyanCavanaugh** [comment]
@mjbvz is there any editor-side support for this?

**Comment by mjbvz** [comment]
We don't have a UI for managing suggestions/actions. For now, this would likely be provided using a setting that VS Code could pass to TS

I believe there are two parts to this request:

- Disabling specific suggestion diagnostics #28825 This seems like the main point of this issue

- Disabling specific fixes / refactorings. Specifically refactorings that do not require any selection. We should get more user feedback before investigating this

**Comment by themattsterr** [comment]
I agree with @heroboy. I want the lightbulb to be displayed, but it is always suggesting I change modifiers to final where possible. 

I'm using the Language Support for Java(TM) by Red Hat extension.

**Comment by heroboy** [comment]
like this
```
"editor.codeActionsOnSave": [
    "source.organizeImports",
    "source.fixAll"
]
```
what about:
```
"editor.codeActionsNotShow": [
    "source.organizeImports",
    "source.fixAll"
]
```

**Comment by orassayag** [comment]
I would like an option to remove this:
![Untitled](https://user-images.githubusercontent.com/5759067/102494902-122e1900-407e-11eb-8254-2e77f5fa86af.png)

**Comment by JakeJohnson05** [comment]
Is there any progress with this feature? I would like to remove SOME but not all VSCode suggestions.

**Comment by PowerSheetAI** [comment]
This would help avoid the auto deletion of unreachable code like occurs with Prettier + ESLint in VSCode, if could just disable that single code action, while still allowing other Prettier formatting to occur, like mentioned in https://github.com/microsoft/vscode/issues/109530.
